src/shared/dissect-image.c

```diff
@@ -3224,6 +3225,16 @@ int mount_image_privately_interactively(
         if (r < 0)
                 return log_error_errno(r, "Failed to relinquish DM and loopback block devices: %m");
 
+        if (ret_dir_fd) {
+                _cleanup_close_ int dir_fd = -EBADF;
+
+                dir_fd = open(created_dir, O_CLOEXEC|O_PATH|O_DIRECTORY);
+                if (dir_fd < 0)
+                        return log_error_errno(errno, "Failed to open mount point directory: %m");
+
+                *ret_dir_fd = TAKE_FD(dir_fd);
+        }
+
         *ret_directory = TAKE_PTR(created_dir);
```

Comment:
smells as if `ret_directory` return param should be optional then too so that either is fd or path as return can be requested.

Reply:
The caller always needs to have the directory so it can be unlinked when done with the mount as we always create a new temporary directory which needs to be cleaned up.

Reply:
ok